src/insert-extension.js

```diff
@@ -34,14 +34,25 @@ export function resetExtensionScriptInsertedOrPresentForTesting() {
 /**
  * Check script info in HTML head and make update if necessary
  * @param {!Window} win
- * @param {!Element} element
  * @param {string} extension
+ * @param {boolean} opt_ignoreElementExistenceCheck if true, will not check and
+ *    require that extension element exist in document
  */
-export function insertAmpExtensionScript(win, element, extension) {
+export function insertAmpExtensionScript(
+    win, extension, opt_ignoreElementExistenceCheck) {
   if (extension == 'amp-embed') {
     extension = 'amp-ad';
   }
-  if (isAmpExtensionScriptRequired(win, element, extension)) {
+  if (!opt_ignoreElementExistenceCheck) {
```

Comment:
No, actually.  We also use it in [AmpA4A#addCreativeExtensions_](https://github.com/google/amphtml/blob/a4a-core/extensions/amp-a4a/0.1/amp-a4a.js#L696) to copy any extensions that a validated AMP ad requests to the host doc head.  The ad creative at that point has not been rendered in the doc and, when it is, it's rendered in shadow DOM, so the `querySelector` won't find them.